src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,322 @@
+import Datepicker from '../../components/datepicker/Datepicker';
+import { PrimaryButton, SecondaryButton } from '../../components/button/Button';
+import { ErrorToast, SuccessToast } from '../../components/notification/Toast';
+import OpeningPeriodDescription from '../description/OpeningPeriodDescription';
+import TimeSpan from '../time-span/TimeSpan';
+import './OpeningPeriodForm.scss';
+
+type OpeningPeriodDataBase = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string | undefined;
+  openingPeriodEndDate: string | undefined;
+};
+
+interface OpeningPeriodFormData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanFormFormat> | {}[];
+}
+
+export interface OpeningPeriodApiData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanApiFormat>;
+}
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+type NotificationTexts = {
+  label: string;
+  description: string;
+};
+
+function formatTimeSpansToApiFormat(
+  timeSpans: TimeSpanFormFormat[]
+): TimeSpanApiFormat[] {
+  return timeSpans.map((timeSpan) => {
+    return {
+      description: {
+        fi: timeSpan?.description,
+        sv: null,
+        en: null,
+      },
+      end_time: `${timeSpan.endTime}:00`,
+      start_time: `${timeSpan.startTime}:00`,
+      resource_state: timeSpan.resourceState,
+      weekdays: timeSpan.weekdays.reduce(
+        (acc: Array<number>, currentValue: boolean, currentIndex: number) => {
+          if (currentValue) {
+            acc.push(currentIndex + 1);
+            return acc;
+          }
+          return acc;
+        },
+        []
+      ),
+    };
+  });
+}
+
+const dropMilliseconds = (time: string): string => time.slice(0, -3);
+const getLowestWeekdayNumber = (timeSpan: TimeSpanApiFormat): number =>
+  timeSpan && timeSpan.weekdays ? timeSpan.weekdays.sort()[0] : 0;
+
+function formatApiTimeSpansToFormFormat(
+  apiTimeSpans: TimeSpanApiFormat[]
+): TimeSpanFormFormat[] {
+  return apiTimeSpans
+    .sort((a: TimeSpanApiFormat, b: TimeSpanApiFormat) => {
+      return getLowestWeekdayNumber(a) - getLowestWeekdayNumber(b);
+    })
+    .map((apiTimeSpan) => {
+      const weekdays: boolean[] = Array(7)
+        .fill(false)
+        .map((current, index): boolean =>
+          apiTimeSpan.weekdays
+            ? apiTimeSpan.weekdays.includes(index + 1)
+            : false
+        );
+
+      return {
+        id: apiTimeSpan.id,
+        description: apiTimeSpan.description?.fi || '',
+        startTime: apiTimeSpan.start_time
+          ? dropMilliseconds(apiTimeSpan.start_time)
+          : '',
+        endTime: apiTimeSpan.end_time
+          ? dropMilliseconds(apiTimeSpan.end_time)
+          : '',
+        resourceState: apiTimeSpan.resource_state || ResourceState.OPEN,
+        weekdays: weekdays as FormWeekdays,
+      };
+    });
+}
+
+const formatResourcePeriodOptionsToFormFormat = (
```

Comment:
Should this be named __formatResourceStateOptionsToFormFormat__ instead?

Reply:
I'll rename

Reply:
Renamed